### ISSUE TITLE
redmine 5

### ISSUE THREAD

**Issue by ashrafalzyoud** [MayamaTakeshi/redmine_rt#9]
its compitable with redmine 5 and ruby 3?
its can work in offline local network without internet

**Comment by MayamaTakeshi** [comment]
I tested now upgrading my redmine installation to version 5 but it didn't work. 

When I tried to start the app I got:
```
[root@red-dev redmine]$ bundle exec rails server -e development -b 0.0.0.0
... ABRIDGED ...
        22: from /var/lib/gems/2.7.0/gems/railties-6.1.6/lib/rails/initializable.rb:32:in `instance_exec'
        21: from /var/lib/gems/2.7.0/gems/railties-6.1.6/lib/rails/application/finisher.rb:124:in `block in <module:Finisher>'
        20: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/reloader.rb:88:in `prepare!'
        19: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:105:in `run_callbacks'
        18: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `invoke_before'
        17: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `each'
        16: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:512:in `block in invoke_before'
        15: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:199:in `block in halting'
        14: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:603:in `block in default_terminator'
        13: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:603:in `catch'
        12: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:604:in `block (2 levels) in default_terminator'
        11: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
        10: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:427:in `block in make_lambda'
         9: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/callbacks.rb:427:in `instance_exec'
         8: from /usr/local/src/git/redmine/lib/redmine/plugin_loader.rb:108:in `block in load'
         7: from /usr/local/src/git/redmine/lib/redmine/plugin_loader.rb:108:in `each'
         6: from /usr/local/src/git/redmine/lib/redmine/plugin_loader.rb:31:in `run_initializer'
         5: from /usr/local/src/git/redmine/lib/redmine/plugin_loader.rb:31:in `load'
         4: from /usr/local/src/git/redmine/plugins/redmine_rt/init.rb:33:in `<top (required)>'
         3: from /var/lib/gems/2.7.0/gems/activesupport-6.1.6/lib/active_support/dependencies/zeitwerk_integration.rb:51:in `require_dependency'
         2: from /var/lib/gems/2.7.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
         1: from /var/lib/gems/2.7.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'
/var/lib/gems/2.7.0/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require': cannot load such file -- redmine_rt/channels/application_cable/connection (LoadError)
```
Then checking the details about 5.0.0 it is explained that plugins will required adjustments to work with it:
  https://www.redmine.org/news/135

However, I currently don't have time to work on this.
So for now I just updated the redmine_rt README page to inform about this limitation.
I will leave this issue open as I eventually might be able to work on it.

**Comment by MayamaTakeshi** [comment]
Regarding your question:
  "its can work in offline local network without internet"
Yes, as long as the redmine server can be accessed by the other machines, the plugin will work (using redmine 4 of course).

**Comment by MayamaTakeshi** [comment]
I had some time to work on this today.
I could make it work with Redmine 5. It is in the branch: zeitwerk_adjustments.
I still need to clean up the code and confirm it keeps working with Redmine 4 before merging it with the master branch but it should be usable already.

Regarding ruby3, I am currently testing with Debian 11 which comes with ruby2.7 so I don't know if it will work or not.

**Comment by ashrafalzyoud** [comment]
Thx for your time and help
We wait u

**Comment by MayamaTakeshi** [comment]
Hi, final changes were committed.
Please refer to updated installation instructions at https://github.com/MayamaTakeshi/redmine_rt

Thanks.

**Comment by MayamaTakeshi** [comment]
Hi, I verified that for redmine 5 the plugin was only working if started in development mode. If production mode was used, redmine 5 would fail to start. This was corrected by #12.
